### ISSUE TITLE
Explore auto-generating api_version docs

### ISSUE THREAD

**Issue by drevell** [abcxyz/abc#462]
### TL;DR

Context: #461

As an abc maintainer, I want to make sure that all supported api_versions are documented in README.md. We want to avoid a situation where we release an api_version but forget to document it.

Possible solution idea: create a simple Go program that runs during CI. This program will fail the CI process if it detects an undocumented, released api_version. The program will run these steps:
 - Look at the list of releases in decode.go to find the set of released `api_version`s (the ones with `unreleased==false`).
 - Look at README.md to find the [table](https://github.com/abcxyz/abc//?tab=readme-ov-file#list-of-api_versions) that documents each `api_version`.
 - Make sure each api_version from step 1 has a table row from step 2, and the table row looks like real docs (the table shouldn't just say "unreleased" or "not released yet").
